rosidl_generator_c/src/string_functions.c

```diff
@@ -84,12 +82,15 @@ rosidl_generator_c__String__assignn(
   if (n == SIZE_MAX) {
     return false;
   }
-  char * data = (char *)malloc(n + 1);
+  char * data = realloc(str->data, n + 1);
   if (!data) {
+    free(str->data);
+    str->data = NULL;
+    str->size = 0;
+    str->capacity = 0;
```

Comment:
The reason I went for this was to protect againt the case where `realloc` fails. But I suppose that, should `realloc` fail, that doesn't mean that the existing string should be freed. So I'll change this to just `return false;` instead.